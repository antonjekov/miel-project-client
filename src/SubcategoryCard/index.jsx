import React from "react";
import { Card } from 'react-bootstrap';
import styles from './index.module.css';

function SubcategoryCard (props){

  const category = props.category

return (
<Card border="warning"  style={{ width: '15rem', height: '15rem'}}>
  <Card.Img variant="top" src={props.subcategory.imageUrl} />
  <Card.Body>    
    <Card.Link className={styles.CardLink} href={`products/${category&&category.name}/${props.subcategory.name}`} >{props.subcategory.name}</Card.Link>
  </Card.Body>
</Card>
);
}

export default SubcategoryCard;