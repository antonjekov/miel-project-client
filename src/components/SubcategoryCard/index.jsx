import React from "react";
import { Card } from 'react-bootstrap';
import styles from './index.module.css';

function SubcategoryCard (props){

return (
<Card border="warning"  style={{ width: '15rem', height: '15rem'}}>  
  <Card.Body>    
    <Card.Link  className={styles.CardLink} href={`/products/${props.category._id}/${props.subcategory._id}`} data-testid="productsHref">
      <Card.Img variant="top" src={props.subcategory.imageUrl} data-testid="subcategoryImage"/>
      <div>
      {props.subcategory.name}
      </div>      
    </Card.Link>
  </Card.Body>
</Card>
);
}

export default SubcategoryCard;