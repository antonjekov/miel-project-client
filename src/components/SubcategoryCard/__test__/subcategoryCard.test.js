import React from "react";
import ReactDOM from "react-dom"
import {render, cleanup} from "@testing-library/react"
import renderer from "react-test-renderer"
import SubcategoryCard from "../"

const category1 = {"_id":"5edbd65159792f2e580f5491","subcategories":["5f046a611f1f5d130809bdd1","5f046c6629be1c3b3c47409a"],"imageUrl":"https://res.cloudinary.com/mielproject/image/upload/v1594107562/it7xvfrjp585dprqk4zm.jpg","name":"honey"};
const subcategory1 = {"_id":"5f046a611f1f5d130809bdd1","products":["5f04a7adaeddbf1fa829908e"],"name":"acacia","category":"5edbd65159792f2e580f5491","description":"Acacia honey has a flower-like aroma and sweet, delicate flavor. Conveniently, acacia honey remains liquid longer and crystallizes much slower than traditional honey.","imageUrl":"https://res.cloudinary.com/mielproject/image/upload/v1594124893/hbezpw1k9wzxgth1yovb.jpg"}

afterEach(cleanup)

it('renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<SubcategoryCard category={category1} subcategory={subcategory1}/>,div)
})

it('renders href and image correctly', ()=>{
    const {getByTestId} = render(<SubcategoryCard category={category1} subcategory={subcategory1}/>)
    expect(getByTestId('productsHref')).toHaveTextContent(subcategory1.name)
    expect(getByTestId('productsHref')).toHaveAttribute('href', `/products/${category1._id}/${subcategory1._id}`)
    expect(getByTestId('subcategoryImage')).toHaveAttribute('src', `${subcategory1.imageUrl}`)
})

it('matches snapshot', ()=>{
    const tree = renderer.create(<SubcategoryCard category={category1} subcategory={subcategory1}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

