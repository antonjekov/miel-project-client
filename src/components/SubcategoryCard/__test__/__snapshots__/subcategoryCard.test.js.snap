// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card border-warning"
  style={
    Object {
      "height": "15rem",
      "width": "15rem",
    }
  }
>
  <img
    className="card-img-top"
    data-testid="subcategoryImage"
    src="https://res.cloudinary.com/mielproject/image/upload/v1594124893/hbezpw1k9wzxgth1yovb.jpg"
  />
  <div
    className="card-body"
  >
    <a
      className="CardLink card-link"
      data-testid="productsHref"
      href="/products/5edbd65159792f2e580f5491/5f046a611f1f5d130809bdd1"
    >
      acacia
    </a>
  </div>
</div>
`;
